spec/models/image_spec.rb

```diff
@@ -0,0 +1,59 @@
+require "rails_helper"
+
+describe Image do
+  describe "class_methods" do
+    before { described_class.image_data }
+
+    it { is_expected.to respond_to(:image_data) }
+
+    specify "sets image_data class variable" do
+      expect(described_class.class_variable_get(:@@image_data)).to be_a(Hash)
+    end
+  end
+
+  describe "methods" do
+    describe "#build_args" do
+      subject { described_class.new.build_args(path) }
+
+      context "with a valid image path" do
+        let(:path) { "media/images/content/hero-images/0003.jpg" }
+
+        specify "returns the correct image path and alt text" do
+          expect(subject).to eql([path, { alt: "Male and female teacher talking in a staff room." }])
+        end
+      end
+
+      context "with a bad image path" do
+        let(:path) { "media/something-really-invalid/abc.pdf" }
+
+        specify "raises an error" do
+          expect { subject }.to raise_error(Image::UnregisteredImageError, /no image information found/)
```

Comment:
Worth having more of an integration test somewhere that iterates over each image and ensures it has alt text specified?

Reply:
These will be caught by the content specs (and/or request specs); if any page is generated using `#image_args` (just blog and hero for the mo) and it fails to find an entry it'll fail with:

```
ActionView::Template::Error:
  no image information found for 'media/images/content/hero-images/0003.jpg', see config/images.yml
```

Reply:
Ah ok cool, I guess its a bit fallible if someone includes an image without using the `Image` class though 🤔

Reply:
Good call, added another commit with a content spec that ensures there's an `alt` attribute on every `img` element.